api/src/pcapi/scripts/achievements/main.py

```diff
@@ -0,0 +1,105 @@
+import logging
+
+from sqlalchemy.orm import contains_eager
+from sqlalchemy.orm import joinedload
+from sqlalchemy.orm import load_only
+
+from pcapi.core.achievements.api import ACHIEVEMENT_NAME_TO_SUBCATEGORY_IDS
+from pcapi.core.achievements.api import _get_booking_achievement
+from pcapi.core.achievements.models import Achievement
+from pcapi.core.bookings.models import Booking
+from pcapi.core.offers.models import Offer
+from pcapi.core.offers.models import Stock
+from pcapi.core.users.models import User
+from pcapi.models import db
+from pcapi.repository import atomic
+
+
+logger = logging.getLogger(__name__)
+
+SUBCATEGORIES_WITH_ACHIEVEMENT = [
+    subcategory_id
+    for subcategory_ids in ACHIEVEMENT_NAME_TO_SUBCATEGORY_IDS.values()
+    for subcategory_id in subcategory_ids
+]
+BATCH_SIZE = 2000
+
+
+def unlock_existing_achievements() -> None:
+    last_user_id = 0
+    has_next_beneficiaries = True
+
+    while has_next_beneficiaries:
+        with atomic():
+            beneficiaries = _get_beneficiaries(last_user_id, BATCH_SIZE)
+
+            booking_with_achievements = _get_achievement_bookings(beneficiaries)
+            _unlock_achievements(booking_with_achievements)
+
+            has_next_beneficiaries = len(beneficiaries) == BATCH_SIZE
+            if has_next_beneficiaries:
+                last_user_id = beneficiaries[-1].id
+                logger.info("Unlocked achievements up to last_user_id = %s", last_user_id)
+            elif beneficiaries:
+                last_user_id = beneficiaries[-1].id
```

Comment:
détail: ici tu if/else que pour le log
```suggestion
            if has_next_beneficiaries:
                logger.info("Unlocked achievements up to last_user_id = %s", last_user_id)
            last_user_id = beneficiaries[-1].id

Reply:
Mais bon c'est un 1-shot script donc verylow logaf